### PR TITLE
Remove legacy parquet reader

### PR DESCRIPTION
**Pull request by raunaqmorarka**
## Description
Remove legacy parquet reader

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# Hive, Hudi, Iceberg, Delta
* Removed legacy parquet reader. The catalog configuration properties `parquet.optimized-reader.enabled` and `parquet.optimized-nested-reader.enabled` are now defunct and must be removed from server configurations. ({issue}`18639`)
```

### CODE REVIEW

File: core/trino-grammar/pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <parent>
         <groupId>io.trino</groupId>
         <artifactId>trino-root</artifactId>
-        <version>423-SNAPSHOT</version>
+        <version>424-SNAPSHOT</version>
```
Comment (findepi) [inline core/trino-grammar/pom.xml:8]: let's have separate PR and merge it faster

**Review by findepi — APPROVED** [review]
(skimmed)

File: core/trino-grammar/pom.xml (hunk 1 above)
Comment (raunaqmorarka) [reply core/trino-grammar/pom.xml:8]: https://github.com/trinodb/trino/pull/18640

**Comment by wendigo** [comment]
Failures seems related

**Pull request merged**